### PR TITLE
DEX-823 Launch load tests via jenkins

### PR DESCRIPTION
**Pull request by ngadiyak**

### CODE REVIEW

File: dex-load/sc1.Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    options {
+        ansiColor('xterm')
+    }
+    parameters {
+        string(name: 'SBT_THREAD_NUMBER', defaultValue: '6', description: '')
+        string(name: 'SEED', defaultValue: 'test-seed', description: '')
+        string(name: 'AN', defaultValue: '60', description: '')
+        string(name: 'RC', defaultValue: '10000', description: '')
+        string(name: 'RT', defaultValue: '6', description: '')
+        string(name: 'AS', defaultValue: 'D', description: '')
```
Comment (vsuharnikov) [inline dex-load/sc1.Jenkinsfile:14]: Let's write a description for each parameter. Now it is not clear, what is AN, RC, ...

File: dex-load/sc1.Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                sh 'rm -rf ./dex-load/*.txt'
+                sh 'git fetch --tags'
+                sh 'find ~/.sbt/1.0/staging/*/waves -type d -name target | xargs -I{} rm -rf {}'
+                sh 'find . -type d -name target | xargs -I{} rm -rf {}'
+                sh 'sbt "cleanAll"'
+            }
+        }
+        stage('Generate ammo file') {
+            steps {
+                sh 'sbt "project dex-load" generateAmmo'
```
Comment (vsuharnikov) [inline dex-load/sc1.Jenkinsfile:42]: Or simpler:
```
sh 'sbt "dex-load/generateAmmo"'
```

File: dex-load/sc2.Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    options {
+        ansiColor('xterm')
+    }
+    parameters {
+        string(name: 'SBT_THREAD_NUMBER', defaultValue: '6', description: '')
+        string(name: 'SEED', defaultValue: 'test-seed', description: '')
+        string(name: 'AN', defaultValue: '60', description: '')
+        string(name: 'RC', defaultValue: '10000', description: '')
+        string(name: 'RT', defaultValue: '6', description: '')
+        string(name: 'AS', defaultValue: 'D', description: '')
```
Comment (vsuharnikov) [inline dex-load/sc2.Jenkinsfile:14]: Same

File: dex-load/sc2.Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                sh 'rm -rf ./dex-load/*.txt'
+                sh 'git fetch --tags'
+                sh 'find ~/.sbt/1.0/staging/*/waves -type d -name target | xargs -I{} rm -rf {}'
+                sh 'find . -type d -name target | xargs -I{} rm -rf {}'
+                sh 'sbt "cleanAll"'
+            }
+        }
+        stage('Generate ammo file') {
+            steps {
+                sh 'sbt "project dex-load" generateAmmo'
```
Comment (vsuharnikov) [inline dex-load/sc2.Jenkinsfile:42]: Same

File: dex-load/sc3.Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    options {
+        ansiColor('xterm')
+    }
+    parameters {
+        string(name: 'SBT_THREAD_NUMBER', defaultValue: '6', description: '')
+        string(name: 'SEED', defaultValue: 'test-seed', description: '')
+        string(name: 'AN', defaultValue: '60', description: '')
+        string(name: 'RC', defaultValue: '10000', description: '')
+        string(name: 'RT', defaultValue: '6', description: '')
+        string(name: 'AS', defaultValue: 'D', description: '')
```
Comment (vsuharnikov) [inline dex-load/sc3.Jenkinsfile:14]: Same

File: dex-load/sc3.Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+                sh 'sbt "cleanAll"'
+            }
+        }
+        stage('Generate feeder file') {
+            steps {
+                sshagent (credentials: ['buildagent-matcher']) {
+                    sh "scp buildagent-matcher@${LOADGEN}:/home/buildagent-matcher/key.txt ./dex-load"
+                    sh "scp buildagent-matcher@${LOADGEN}:/home/buildagent-matcher/pairs.txt ./dex-load"
+                }
+                sh 'sbt "project dex-load" generateFeeder'
```
Comment (vsuharnikov) [inline dex-load/sc3.Jenkinsfile:46]: Same

File: dex-load/sc4.Jenkinsfile [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    options {
+        ansiColor('xterm')
+    }
+    parameters {
+        string(name: 'SBT_THREAD_NUMBER', defaultValue: '6', description: '')
+        string(name: 'SEED', defaultValue: 'test-seed', description: '')
+        string(name: 'AN', defaultValue: '60', description: '')
+        string(name: 'RC', defaultValue: '10000', description: '')
+        string(name: 'RT', defaultValue: '6', description: '')
+        string(name: 'AS', defaultValue: 'D', description: '')
```
Comment (vsuharnikov) [inline dex-load/sc4.Jenkinsfile:14]: Same

File: dex-load/sc4.Jenkinsfile [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+                sh 'rm -rf ./dex-load/*.txt'
+                sh 'git fetch --tags'
+                sh 'find ~/.sbt/1.0/staging/*/waves -type d -name target | xargs -I{} rm -rf {}'
+                sh 'find . -type d -name target | xargs -I{} rm -rf {}'
+                sh 'sbt "cleanAll"'
+            }
+        }
+        stage('Generate ammo file') {
+            steps {
+                sh 'sbt "project dex-load" generateAmmo'
```
Comment (vsuharnikov) [inline dex-load/sc4.Jenkinsfile:42]: Same

File: dex-load/sc4.Jenkinsfile [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+            }
+        }
+        stage('Generate feeder file') {
+            steps {
+                sshagent (credentials: ['buildagent-matcher']) {
+                    sh "scp buildagent-matcher@${LOADGEN}:/home/buildagent-matcher/key.txt ./dex-load"
+                    sh "scp buildagent-matcher@${LOADGEN}:/home/buildagent-matcher/pairs.txt ./dex-load"
+                }
+                sh 'sbt "project dex-load" generateFeeder'
```
Comment (vsuharnikov) [inline dex-load/sc4.Jenkinsfile:51]: Same

File: dex-load/sc2.Jenkinsfile [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+            }
+        }
+        stage('Generate feeder file') {
+            steps {
+                sshagent (credentials: ['buildagent-matcher']) {
+                    sh "scp buildagent-matcher@${LOADGEN}:/home/buildagent-matcher/key.txt ./dex-load"
+                    sh "scp buildagent-matcher@${LOADGEN}:/home/buildagent-matcher/pairs.txt ./dex-load"
+                }
+                sh 'sbt "project dex-load" generateFeeder'
```
Comment (vsuharnikov) [inline dex-load/sc2.Jenkinsfile:51]: Same

File: dex-ws-load/build.sbt [hunk 11]
```diff
@@ -0,0 +1,17 @@
+enablePlugins(GatlingPlugin)
+
+scalaVersion := "2.12.10"
```
Comment (vsuharnikov) [inline dex-ws-load/build.sbt:3]: ~~Interesting, why we can't use 2.13.2 here as in other modules in the project?~~
I've found, it is because of Gatling.

File: dex-ws-load/build.sbt [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+
+scalacOptions := Seq(
+  "-encoding", "UTF-8", "-target:jvm-1.8", "-deprecation",
+  "-feature", "-unchecked", "-language:implicitConversions", "-language:postfixOps")
+
+libraryDependencies += "io.gatling.highcharts" % "gatling-charts-highcharts" % "3.3.1" % "test,it"
+libraryDependencies += "io.gatling"            % "gatling-test-framework"    % "3.3.1" % "test,it"
+libraryDependencies += "org.scalactic" %% "scalactic" % "3.0.8"
+libraryDependencies += "org.scalaj" % "scalaj-http_2.11" % "2.3.0"
```
Comment (vsuharnikov) [inline dex-ws-load/build.sbt:12]: It looks like there are 2.4.2 that supports not only Scala 2.11 https://search.maven.org/search?q=scalaj-http
Consider:
```
"org.scalaj" %% "scalaj-http" % "2.4.2"
```

File: dex-ws-load/build.sbt [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+
+
+
+libraryDependencies += "com.wavesplatform" % "wavesj" % "0.16.0"
+libraryDependencies += "org.scalatest" %% "scalatest" % "3.0.8" % "test"
```
Comment (vsuharnikov) [inline dex-ws-load/build.sbt:17]: There is a newline required.

Also we store dependencies in `Dependencies.scala` to update them easier. Also note: if you really need Scala 2.12, then you should write (in Dependencies.scala) for your dependencies:
```
"foo" % "bar_2.12" % "x.y.z"
```
 instead of:
```
"foo" %% "bar" % "x.y.z"
```

File: dex-ws-load/project/build.properties [hunk 14]
```diff
@@ -0,0 +1 @@
+sbt.version=1.3.3
```
Comment (vsuharnikov) [inline dex-ws-load/project/build.properties:1]: Is this file really needed?

File: dex-ws-load/project/plugins.sbt [hunk 15]
```diff
@@ -0,0 +1 @@
+addSbtPlugin("io.gatling" % "gatling-sbt" % "3.1.0")
```
Comment (vsuharnikov) [inline dex-ws-load/project/plugins.sbt:1]: I see, it supports only Scala 2.12 :(

File: dex-ws-load/src/test/resources/gatling.conf [hunk 16]
```diff
@@ -0,0 +1,127 @@
+#########################
```
Comment (vsuharnikov) [inline dex-ws-load/src/test/resources/gatling.conf:1]: All options are commented. Why do we need this file?

File: dex-ws-load/src/test/resources/recorder.conf [hunk 17]
```diff
@@ -0,0 +1,37 @@
+recorder {
```
Comment (vsuharnikov) [inline dex-ws-load/src/test/resources/recorder.conf:1]: Same questions for this file

**Review by vsuharnikov — CHANGES_REQUESTED** [review]

File: dex-ws-load/src/test/resources/gatling.conf (hunk 16 above)
Comment (ngadiyak) [reply dex-ws-load/src/test/resources/gatling.conf]: deleted

File: dex-ws-load/src/test/resources/recorder.conf (hunk 17 above)
Comment (ngadiyak) [reply dex-ws-load/src/test/resources/recorder.conf]: deleted

File: dex-ws-load/project/build.properties (hunk 14 above)
Comment (ngadiyak) [reply dex-ws-load/project/build.properties]: deleted

File: dex-load/sc2.Jenkinsfile (hunk 3 above)
Comment (ngadiyak) [reply dex-load/sc2.Jenkinsfile]: fixed

File: dex-load/sc1.Jenkinsfile (hunk 1 above)
Comment (ngadiyak) [reply dex-load/sc1.Jenkinsfile]: fixed

File: dex-load/sc1.Jenkinsfile (hunk 2 above)
Comment (ngadiyak) [reply dex-load/sc1.Jenkinsfile]: fixed

**Review by vsuharnikov — APPROVED** [review]
